rfc/egress-traffic.md

```diff
@@ -0,0 +1,200 @@
+# RFC: Egress Traffic Tracking
+
+## Authors
+
+- Felipe Forbeck
+
+## Status
+
+- Proposed
+
+## Context
+
+Storacha Network aims to implement a scalable and automated mechanism to track egress traffic for customer billing purposes. The objective is to update Stripe’s API with relevant egress data for each customer, ensuring accurate billing. This RFC outlines two potential approaches to achieve this goal, each with its own set of trade-offs.
+
+## Problem
+
+Currently, there is no established automated process to track and associate egress traffic per customer and report this usage to Stripe for billing. A solution is required to automate this process for seamless, scalable, and accurate billing based on egress traffic.
+
+## Proposal
+
+We propose two alternatives for tracking and updating egress billing traffic:
```

Comment:
Stepping back - do you have an idea how you're going to log the right amount of egress? Count bytes? What happens when the connection is closed - can that be handled in a worker?

Do cloudflare have logs you could just crawl?

Reply:
I don't think cloudflare has historical logs like this, but I could be mistaken

Reply:
Also updated the RFC to handle the closed connection issue in another iteration using the Transformation Streams as suggested by @alanshaw .